src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -193,16 +228,31 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
             {
                 return Failure(
                     factory,
+                    logger,
+                    context.Logger,
                     Strings.MSBuildSDKDirectoryNotFound,
                     msbuildSdkDir);
             }
 
             return factory.IndicateSuccess(msbuildSdkDir, netcoreSdkVersion, propertiesToAdd, itemsToAdd, warnings);
         }
 
-        private static SdkResult Failure(SdkResultFactory factory, string format, params object[] args)
+        private static SdkResult Failure(SdkResultFactory factory, ResolverLogger logger, SdkLogger sdkLogger, string format, params object[] args)
         {
-            return factory.IndicateFailure(new[] { string.Format(format, args) });
+            string error = string.Format(format, args);
+            logger.LogMessage($"Failed to resolve SDK: {error}");
+
+            CopyLogMessages(logger, sdkLogger);
+
+            return factory.IndicateFailure(new[] { error });
+        }
+
+        private static void CopyLogMessages(ResolverLogger source, SdkLogger destination)
```

Comment:
This seems like it doesn't care if a message has already been logged in the destination logger. Is there any way we can verify that the messages haven't been logged already? Like glancing above, it seems like this could be called twice if we hit line 204 then 229.